### ISSUE TITLE
Discord not picking up Runelite activity with Bolt on Steamdeck

### ISSUE THREAD

**Issue by Djakamoe** [Adamcake/Bolt#88]
Hi, guys. Thanks for your work, but...

There is a small problem I'm having. 

Using version 0.10.0 via Discover install I can't get Discord to pickup Runelite activity when I launch it with the Bolt launcher. 

Launching Runelite by itself it works, but using the launcher like I have to in order to log in it will not. 

There _might_ be a workaround here by adding the Runelite JAR that the Bolt launcher uses to steam, but I can't seem to find that either.

Adding the Bolt launcher to steam shows activity that I'm playing Bolt, but that's not really helpful information like Runelite displays.

If there is anything you guys need from me for this just let me know. I'll check in on this pretty often, as I'd like to get this solved quickly. Thanks!

**Comment by Adamcake** [comment]
For runelite and discord to be visible to each other in flatpak's sandbox, you need to do this: https://wiki.archlinux.org/title/Discord#Enabling_rich_presence_on_Flatpak

**Comment by TheInfamousAlk** [comment]
That doesn't do anything for me, as I'm not using the flatpak version of discord. Rich presence works for other apps, just not RuneLite launched with Bolt.

**Comment by Adamcake** [comment]
If you're using flatpak bolt but not flatpak discord, then Discord should be creating a socket called `discord-ipc-0` in the actual runtime dir but Bolt wouldn't be able to see it. If you check if that file exists and set Bolt's permissions in flatseal to be able to see it, it should work then.

**Comment by TheInfamousAlk** [comment]
I can't see that, see: 

```
alk@pop-os:~$ ls $XDG_RUNTIME_DIR/app/
ch.protonmail.protonmail-bridge  com.adamcake.Bolt  com.core447.StreamController  com.github.tchx84.Flatseal  io.freetubeapp.FreeTube  io.github.amit9838.mousam  org.mozilla.Thunderbird

```

No Discord in there. Where else would it be? Discord is installed through apt not flatpak. Other games seem to work with discord RPC so there must be another socket they are using.

**Comment by Adamcake** [comment]
`ls $XDG_RUNTIME_DIR`, it should be in there. the `app` directory belongs to flatpak only.

**Comment by TheInfamousAlk** [comment]
You're right, it is in there. However, neither Bolt nor the Runelite instance it creates show up in discord.

**Comment by Adamcake** [comment]
Yes, that's because Bolt is sandboxed so it can't see that directory. To give it permission to see all of XDG_RUNTIME_DIR, you can use Flatseal and add an entry like this:

![image](https://github.com/user-attachments/assets/a65fed1b-75a2-4775-af42-94f6fa396e2e)

Alternatively, you could symlink the discord-ipc file to the location that Bolt is looking for it. But that's a little more complex since you'd have to do it every time the system starts.

**Comment by TheInfamousAlk** [comment]
I added that, it still doesn't pick it up in Discord. 
![image](https://github.com/user-attachments/assets/07bd58bf-f970-4722-94ca-3875e746c576)

**Comment by Adamcake** [comment]
Oh, sorry. I should have checked the docs first:

> Subdirectories of the XDG_RUNTIME_DIR defined by the freedesktop.org Base Directory Specification. Note that xdg-run on its own is not supported.

So you need to change it to `xdg-run/discord-ipc-0` instead.

**Comment by TheInfamousAlk** [comment]
AYY there it is. Thank you so much for your help. 

I noticed there were 2 entries in there by default, would you be able to include `xdg-run/discord-ipc-0` in there as well, as a default for users in my situation?

**Comment by Adamcake** [comment]
That's a good idea. Your situation is an unusual one, but I can't see any downsides to doing that. I'll do it next time I put a release on flatpak.

Anyway - I realise you're not the original creator of this issue, but since they haven't replied since November I'm going to close this anyway.

**Comment by TheInfamousAlk** [comment]
Okay, this has actually stopped working again. I have updated discord recently, but not any flatseal settings or bolt itself. I still have xdg-run/discord-ipc-0 specified in flatseal. discord-ipc-0 is still visible in $XDG_RUNTIME_DIR/ so I don't know what's going on. I don't know if it stopped working before or after discord's update, as I haven't played runescape at all in the last week.

**Comment by Adamcake** [comment]
Hmm, you could look at the logs in `.runelite` to see where it's looking for the IPC file and whether it's finding it. I think it says something about that.

**Comment by TheInfamousAlk** [comment]
Okay it works after a restart with no other changes, and I restarted several times and let it run for hours and tried a few different things and couldn't get it to break again... so I'll just pretend that didn't happen and hope it works haha.
